enable cors to access rss feeds corss origin

sean1588 opened the pull request:
This PR enables us to access the rss feeds cross origin, since I need to access them from the console UI. I tried this out in a dev environment by standing up cloudfront and the bucket and confirming I can fetch from the browser console.  This will allow origins from any subdomain of pulumi.com or any subdomain of pulumi-dev.io

cnunciato commented on scripts/cors/cors.json:
```diff
@@ -0,0 +1,13 @@
+{
+  "CORSRules": [
+    {
+      "AllowedOrigins": [
```
It's probably worth a comment here explaining that this is consumed by the Service. Also, do we somehow suppress this functionality in self-hosted contexts? Since those won't be running on these domains, we should probably just make sure the UI won't break when it can't load this info.

sean1588 replied:
ahhh yeah that is a very good point! I imagine we would want to show to self hosted if possible as they are pulumi users so the events will likely be relevant to them.

sean1588 replied:
I am going to go ahead and hide the events dashboard card from self hosted folks for now. This is something we had wanted to rollout quickly to see what the response is like first and then plan to iterate on it more in the future, so its fine for now if we do not show to them.

cnunciato commented on scripts/cors/cors.json:
```diff
@@ -0,0 +1,13 @@
+        "https://*.pulumi.com",
+        "https://*.pulumi-dev.io"
+      ],
+      "AllowedHeaders": ["Authorization"],
```
Is this saying that auth headers are OK to send with these requests? If so, we probably should not be doing this -- but apologies if I'm reading this wrong.

sean1588 replied:
yes you are right. thanks christian, we should not be doing this.

cnunciato reviewed (changes requested):
Marking with "request changes" only to be on the safe side WRT that auth-header setting.

sean1588 commented on scripts/sync-and-test-bucket.sh:
```diff
@@ -99,6 +99,9 @@ aws s3 cp "$metadata_file" "${destination_bucket_uri}/metadata.json" --region "$
 set_bucket_for_commit "$(git_sha)" "$destination_bucket" "$(aws_region)"
 
+# Set cors configuration on bucket.
+aws s3api put-bucket-cors --bucket "$destination_bucket" --cors-configuration file://scripts/cors/cors.json
```
ok, i'll add the region. this worked fine for me locally as is. though could be because I have the `AWS_REGION` env var set, so it took that.

sean1588 commented:
@cnunciato - I removed the authorization header from the allowed list and also added a couple extra entries to the allowed origins list to include localhost:3000 and localhost:4200, so that we can render this when running locally. also, i'm going to make the adjustment in the console code to exclude this from self-hosted users for now. Let me know what you think.
